pkg/controller/controllerutil/controllerutil.go

```diff
@@ -60,6 +60,10 @@ func SetControllerReference(owner, object metav1.Object, scheme *runtime.Scheme)
 		return fmt.Errorf("%T is not a runtime.Object, cannot call SetControllerReference", owner)
 	}
 
+	if owner.GetNamespace() != "" && owner.GetNamespace() != object.GetNamespace() {
+		return fmt.Errorf("cross-namespace owner references are disallowed, owner's namespace %s, obj's namespace %s", owner.GetNamespace(), object.GetNamespace())
```

Comment:
This happens to (accidentally?) also cover the case of "cluster-scoped resource must not have a namespace-scoped owner". However in that case the error is confusing, because its not a cross-namespace ref.

Would you mind returning a different error message in case `object.GetNamesapce() == ""`?

Reply:
my intention was to cover both cases. I had separated them now, it's more clear. thx for suggestion.